backend/src/rss.rs

```diff
@@ -0,0 +1,192 @@
+                rss_link,
+                video_items
+            )
+        })
+        .map_err(|error| Error::msg(format!("Error generating video items: {:?}", error)));
+        
+    rss_content
+}
+
+/// Generates the single video items of a series in Tobira for inclusion in an RSS feed.
+async fn generate_video_items(
+    db: &Client,
+    series_oc_id: &str,
+    series_title: &str,
+    rss_link: &str,
+    tobira_url: &str,
+) -> Result<String, Error> {
+    let events = gather_event_data(&db, &series_oc_id).await?;
+
+    let mut video_items = String::new();
+
+    for event in events {
+        let mut buf = [0; 11];
+        let tobira_event_id = event.id.to_base64(&mut buf);
+        let event_link = format!("{}/!v/{}", tobira_url, tobira_event_id);
+
+        let preferred_track = choose_preferred_track(event.tracks).unwrap();
+
+        let enclosure_url = &preferred_track.uri;
+        let mimetype = &preferred_track.mimetype.unwrap();
+
+        let item = format!(
+            r#"
+            <item>
+              <title>{}</title>
+              <link>{}</link>
+              <description>{}</description>
+              <dc:creator>{}</dc:creator>
+              <enclosure url="{}" type="{}" length="0" />
+              <guid isPermaLink="true">{}</guid>
+              <pubDate>{}</pubDate>
+              <source url="{}">{}</source>
+            </item>"#,
+            event.title,
+            event_link,
+            event.description.unwrap_or_default(),
+            event.creators.join(", "),
+            enclosure_url,
+            mimetype,
+            event_link,
+            event.created.to_rfc2822(),
+            rss_link,
+            series_title,
+        );
+
+        video_items.push_str(&item);
+    }
+
+    Ok(video_items)
+}
+
+// Gathers the needed event data for video items to include in an RSS feed of a series in Tobira.
+async fn gather_event_data(db: &Client, series_id: &str) -> Result<Vec<Event>, Error> {
+    let query = "select id, title, description, created, creators, tracks from events where part_of = $1";
+    let rows = db.query(query, &[&series_id]).await?;
+
+    let mut events = Vec::new();
+
+    for row in rows {
+        let id = row.get::<_, Key>("id");
+        let title = row.get("title");
+        let description = row.get("description");
+        let created = row.get("created");
+        let creators = row.get("creators");
+        let tracks = row.get("tracks");
+
+        let event = Event {
+            id,
+            title,
+            description,
+            created,
+            creators,
+            tracks,
+        };
+
+        events.push(event);
+    }
+
+    Ok(events)
+}
+
+/// This returns a track that:
+/// a) is the `presentation`` track.
+/// Defaults to any track meeting the b) criteria if none there is no `presentation` track.
+/// b) has a resolution that is closest to full hd.
+fn choose_preferred_track(tracks: Vec<EventTrack>) -> Option<EventTrack> {
```

Comment:
I suppose one could use [`media:group`](https://www.rssboard.org/media-rss#media-group) to just specify all tracks? The spec sounds very much like this is designed for this purpose of having multiple encodings of the same content. Also, `height` and `width` should then be specified for the `media:content` tag. 

This still leaves the problem of dual stream videos of course. No idea. How does ETH do that?

Reply:
Quoting Olaf's [comment](https://github.com/elan-ev/tobira/issues/791#issuecomment-1534405300) again,
> - Provide Full HD as a standard
> - Dual video: Only provide "presentation"

That's what I tried doing here and think I do also prefer not specifying all tracks. But as always, that's not a super hard opinion.

Reply:
I think Olaf might be talking about the two different RSS files shown in his screenshot? Why would you prefer not to specify all tracks?

Reply:
I was talking about resolution (thinking about only a single-flavor video). In our share menu, we would only link to a single RSS file. And that RSS file just mentions all different qualities. That was what I suggested. But yeah, one certainly has to try how RSS readers display that.

Reply:
I haven't tried this yet so it is still missing in my latest push.

Reply:
Included now, but we still need no filter tracks for flavor and resolution for the `<enclosure>` element. I'm also filtering for `presentation` tracks for the `<media:content>` things in `<media:group>` but this could also list include every track. I _think_ the media things are solely for download purposes, while the `<enclosure>` is used in _some_ rss readers to embed the video... needless to say, rss and this whole PR is giving me headaches. You as well, I presume. Also, deployment failing. Yay.

Reply:
Ah yeah, that's better already. I looked into this again. I am still confused but I think:
- It is ok to not have `enclosure`? Or have you tried this and some RSS readers don't work then? You said that the `media` things are mostly for download so mh...
- I think it's possible to have multiple `<media:group>`s -> we could have one per flavor?

Dunno uhg, I really don't know.

Reply:
> It is ok to not have enclosure?

It's a requirement for apple podcasts :(

> I think it's possible to have multiple <media:group>s -> we could have one per flavor?

Sure, that works.